tensorflow/core/kernels/lstm_ops.h

```diff
@@ -0,0 +1,265 @@
+#ifndef TENSORFLOW_KERNELS_LSTM_OPS_H_
+#define TENSORFLOW_KERNELS_LSTM_OPS_H_
+
+#include "third_party/eigen3/unsupported/Eigen/CXX11/Tensor"
+#include "tensorflow/core/framework/tensor_types.h"
+#include "tensorflow/core/platform/types.h"
+#include "tensorflow/stream_executor/stream.h"
+
+namespace tensorflow {
+
+#if GOOGLE_CUDA
+
+namespace {
+template <typename T>
+perftools::gputools::DeviceMemory<T> AsDeviceMemory(const T* cuda_memory) {
+  perftools::gputools::DeviceMemoryBase wrapped(const_cast<T*>(cuda_memory));
+  perftools::gputools::DeviceMemory<T> typed(wrapped);
+  return typed;
+}
+}  // namespace
+
+#endif  // GOOGLE_CUDA
+
+namespace functor {
+
+template <typename Device, bool USE_CUBLAS>
```

Comment:
instead of defining USE_CUBLAS, I would partially specialize the implementation of these operators by their device only, since USE_CUBLAS is equivalent to Device=GPUDevice